internal/gatewayapi/backendtrafficpolicy.go

```diff
@@ -125,15 +136,24 @@ func (t *Translator) ProcessBackendTrafficPolicies(backendTrafficPolicies []*egv
 			res = append(res, policy)
 
 			// Negative statuses have already been assigned so its safe to skip
-			gateway := resolveBTPolicyGatewayTargetRef(policy, gatewayMap)
+			gateway := resolveBTPolicyGatewayTargetRef(policy, t.GatewayControllerName, gatewayMap)
 			if gateway == nil {
 				continue
 			}
 
-			t.translateBackendTrafficPolicyForGateway(policy, gateway, xdsIR)
+			// Use the gatewayclass of gateway directly as its ancestor reference,
+			// since one gateway only associate with one gatewayclass.
+			ancestorRefs := []gwv1a2.ParentReference{
```

Comment:
IMO, we should keep consistency and use `Gateway` as the ancestor if we can, which is suggested by the Gateway API. When a `Status` doesn't directly related to a specific `Gateway`(e.g, in the case of invalid configuration), we can use `GatewayClass` as the ancestor.

Reply:
Modified